middleware/modelExport.js

```diff
@@ -329,7 +329,11 @@ module.exports = {
 
         });
 
-        async.waterfall(asyncFn, function (err, filters, form, list, exportObj) {
+        asyncFn.push(function (filters, form, list, exportObj, cb) {
```

Comment:
@allanchau, you can't see it in this diff, but it looks as though the previous asyncFn function returns `null` as the first callback, which I'm guessing is a space to hold an `err` as per Node.js convention. However your function doesn't provide a spot for it but instead would make `filters` null?

Reply:
Am I missing something?

Reply:
hrmm in the next line I am passing `err` to the callback (which should be undefined).
It is pretty much the same as the block above L322 except it adds labels to the parameters?

Reply:
But that `err` is coming from the `getLabels` call...

Reply:
Yeah so if there is an error, err will be defined and the asyncFn will go straight to L339.
If there is no error, it will skip L338 because err is undefined.

The previous async function just says it will never fail because err is always null.

Reply:
Yeah right, I get you. Makes sense. Thanks.